tests/ui/missing_transmute_annotations.rs

```diff
@@ -0,0 +1,64 @@
+//@aux-build:macro_rules.rs
+
+#![warn(clippy::missing_transmute_annotations)]
+
+#[macro_use]
+extern crate macro_rules;
+
+macro_rules! local_bad_transmute {
+    ($e:expr) => {
+        std::mem::transmute($e)
```

Comment:
Can you special case `_` here in all macros? The type can't always be specified like this automatically

Reply:
In such cases, I think it means the macro should be rewritten instead of us dropping the level of detection.